File: app/src/main/java/com/scatl/uestcbbs/module/magic/adapter/MineMagicAdapter.java
package com.scatl.uestcbbs.module.magic.adapter;

import android.view.View;

import androidx.annotation.NonNull;

import com.chad.library.adapter.base.BaseQuickAdapter;
import com.chad.library.adapter.base.BaseViewHolder;
import com.scatl.uestcbbs.R;
import com.scatl.uestcbbs.entity.MineMagicBean;
import com.scatl.uestcbbs.helper.glidehelper.GlideLoader4Common;

public class MineMagicAdapter extends BaseQuickAdapter<MineMagicBean.ItemList, BaseViewHolder> {
    public MineMagicAdapter(int layoutResId) {
        super(layoutResId);
    }

    @Override
    protected void convert(@NonNull BaseViewHolder helper, MineMagicBean.ItemList item) {
        helper.setText(R.id.item_mine_magic_dsp, item.dsp)
                .setText(R.id.item_mine_magic_name, item.name)
                .setText(R.id.item_mine_magic_weight, "数量：" + item.totalCount + "  重量：" + item.totalWeight)
                .addOnClickListener(R.id.item_mine_magic_use_btn);

        helper.getView(R.id.item_mine_magic_use_btn).setVisibility(item.showUseBtn ? View.VISIBLE : View.GONE);

        GlideLoader4Common.simpleLoad(mContext, item.icon, helper.getView(R.id.item_mine_magic_icon));
    }
}

File: app/src/main/java/com/scatl/uestcbbs/module/magic/adapter/MineMagicAdapter.kt
package com.scatl.uestcbbs.module.magic.adapter

import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import com.scatl.uestcbbs.databinding.ItemMineMagicBinding
import com.scatl.uestcbbs.entity.MineMagicBean
import com.scatl.uestcbbs.helper.PreloadAdapter
import com.scatl.uestcbbs.helper.ViewBindingHolder
import com.scatl.uestcbbs.util.load

/**
 * Created by sca_tl at 2023/6/12 15:20
 */
class MineMagicAdapter: PreloadAdapter<MineMagicBean.ItemList, ItemMineMagicBinding>() {

    override fun getViewBinding(parent: ViewGroup): ItemMineMagicBinding {
        return ItemMineMagicBinding.inflate(LayoutInflater.from(context), parent, false)
    }

    override fun onBindViewHolder(holder: ViewBindingHolder<ItemMineMagicBinding>, position: Int, item: MineMagicBean.ItemList?) {
        super.onBindViewHolder(holder, position, item)
        if (item == null) {
            return
        }

        holder.binding.magicDsp.text = item.dsp
        holder.binding.magicName.text = item.name
        holder.binding.magicWeight.text = "数量：${item.totalCount}  重量：${item.totalWeight}"
        holder.binding.magicIcon.load(item.icon)
        holder.binding.magicUseBtn.visibility = if (item.showUseBtn) View.VISIBLE else View.GONE
    }
}